Sgx how attestion (#18)
aul12/PraktischeITSicherheit-Intel-SGX: Lecture/Attestation.tex

\section{Attestation}
\begin{frame}
    \frametitle{Attestation}
    \begin{itemize}[<+->]
        \item Attestation is used to verify the integrity of an enclave:
        \begin{itemize}[<+->]
            \item Verifies that the code is trustable

>             \item That the enclave Code isn't tampered

Reviewer: ```suggestion
            \item That the enclave code has not been tampered with
```

        \end{itemize}

>         \item For attestation a report will be generated by enclave, which contains the identity, enclaves measurement and a supplied message from other enclave

Reviewer: ```suggestion
        \item For attestation a report will be generated by enclave, which contains the identity, measurement of the enclave and a supplied message from other enclave
```

>         \item Report is signeencrypted by the enclave with a symmetric key between target enclave and SGX Implementation

Reviewer: signencrypted?

Reviewer: Implementation klein

Reviewer: Statt "SGX implementation" "Processor"?

>         \item Attestation is for a data exchange between 2 enclaves on same CPU or via remote 

Reviewer: Was jetzt Data Exchange oder Verify Integrity?

>         \item Attestation is for a verify integrity between 2 enclaves on same CPU or via remote 

Reviewer: ```suggestion
        \item Attestation is used to verify integrity of enclaves on same CPU or via remote 
```

    \end{itemize}
\end{frame}

\begin{frame}
    \frametitle{Local Attestation}
    \begin{columns}
        \begin{column}{0.4\textwidth}
            \begin{enumerate}[<+->]
                \item Enclave A receives the enclave Measure (MRENCLAVE) from enclave B via an established channel

>                 \item Enclave A generates the report for B with MRENCLAVE and send the report back so B can verify with EGETKEY the report

Reviewer: ```suggestion
                \item Enclave A generates the report for B with MRENCLAVE and send the report back so B can verify the report using EGETKEY
```

                \item Enclave B generates the report for A with the MRENCLAVE from A and sends it back to get verified from A
            \end{enumerate}
        \end{column}
        \begin{column}{0.6\textwidth}
            \includegraphics[scale=0.45]{Images/local_attestation.png}
        \end{column}
    \end{columns}
\end{frame}

\begin{frame}
    \frametitle{Remote Attestation}
    \begin{columns}
        \begin{column}{0.4\textwidth}
            \begin{enumerate}[<+->]

>                 \item Application receives an attestion request from Challenger

Reviewer: ```suggestion
                \item Application receives an attestion request from challenger
```
Challenger wurde noch nicht erklärt

>                 \item Report is requested from application

Reviewer: ```suggestion
                \item Application requests report
```

                \item Application receives report
                \item Report is send to quoting enclave (QE)
                \item QE converts local report to remote report (Quote) and sends back

>                 \item Challegner receives quote

Reviewer: ```suggestion
                \item Challenger receives quote
```

                \item Attestation verification service verifies quote
            \end{enumerate}
        \end{column}
        \begin{column}{0.6\textwidth}
            \includegraphics[scale=0.4]{Images/remote_attestation.png}
        \end{column}
    \end{columns}
\end{frame}

\begin{frame}
    \frametitle{Sealing}
    \begin{itemize}[<+->]
        \item Secrets of an enclave gets lost when enclave is closed
        \item Process of encrypting enclave secrets for persistent storage is called sealing
        \item The data can be signed with different keys

>         \item Enclave retrieves the sealing Key using the EGETKEY instruction, which returns a cryptographic key

Reviewer: ```suggestion
        \item Enclave retrieves the sealing key using the EGETKEY instruction, which returns a cryptographic key
```

>         \item Enclave retrieves the seal Key using the EGETKEY instruction, which returns a cryptographical key

Reviewer: ```suggestion
        \item Enclave retrieves the sealing key using the EGETKEY instruction, which returns a cryptographic key
```

>         \item Enclaves can seal data with MRSIGNER as key, so no other enclave is able to read data

Reviewer 2: ```suggestion
        \item Enclaves can seal data with MRENCLAVE as key, so no other enclave is able to read data
```
